### ISSUE TITLE
BUG: Composer package is no longer functional

### ISSUE THREAD

**Issue by johnbillion** [10up/ElasticPress#2727]
**Describe the bug**

ElasticPress is distributed via Packagist at https://packagist.org/packages/10up/elasticpress. Since 4.0, this package is no longer installable via Composer without also installing its developer dependencies and performing a build step for the same reasons outlined in the 4.0 release notes -- the built assets are no longer present in the source repo.

The Packagist integration with GitHub does not allow a custom release asset to be used, it only uses the zip for the tag (`/refs/tags/x.y.z.zip`), so the `elasticpress-x.y.z.zip` asset is useless to Packagist.

**Potential Solution**

I had this exact problem with my Query Monitor plugin when I wanted to remove the built assets from the source repo but retain the Composer package on Packagist. I solved it by using a GitHub action to run the build script, commit the additions, and tag the release with the built files included.

The workflow is here: https://github.com/johnbillion/query-monitor/blob/develop/.github/workflows/build.yml. The subsequent release to w.org still happens with 10up/action-wordpress-plugin-deploy: https://github.com/johnbillion/query-monitor/blob/develop/.github/workflows/deploy-tag.yml. Steps to deploy can be found here: https://github.com/johnbillion/query-monitor/blob/develop/CONTRIBUTING.md#releasing-a-new-version.

The main difference between this approach and the existing https://github.com/10up/ElasticPress/blob/develop/.github/workflows/push-deploy.yml is that with QM I push to a release branch which then gets built as the release, rather than the build happening on the `release` action.

It would be great if the package on Packagist was fixed so ElasticPress could be installed via Composer again.

**Comment by felipeelia** [comment]
Hi @johnbillion, always good to see you here!

I hope this change didn't get you off guard, as it has been advertised in our readme and changelogs for a while. Is there any reason why you couldn't pull the built version from wpackagist.org?

**Comment by johnbillion** [comment]
While I was working on this problem for Query Monitor I originally intended to recommend using the WPackagist repo instead, and I abandoned the package on Packagist in favour of that, but it was pointed out that this is a third party repo that isn't vetted. It's fine for individuals to make the decision to use WPackagist but not ideal for a package provider to recommend it as its preferred repo.

Ultimately I decided to fix the Packagist build and it ended up being not too difficult.
